apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -18,6 +21,10 @@ func (host *BareMetalHost) validateHost() []error {
 		errs = append(errs, err)
 	}
 
+	if err := validateDNSName(host.Spec.BMC.Address); err != nil {
```

Comment:
@hroyrh hardwareutils/bmc package is added in this module. Maybe it would be useful using bmc's functionality for that validation if it is possible.

Reply:
I agree, the validation itself should probably go to the bmc package, if it's not there already